tools/cli/admin_kafka_commands_test.go

```diff
@@ -0,0 +1,975 @@
+	t.Run("successful parse replication task", func(t *testing.T) {
+		wireVal, _ := (&replicator.ReplicationTask{CreationTime: ptr.Int64(123)}).ToWire()
+		encoded := bytes.NewBuffer([]byte{})
+		_ = binary.Default.Encode(wireVal, encoded)
+		data := []byte("Partition: abc, Offset: 0, Key: 123")
+		data = append(data, preambleVersion0)
+		data = append(data, encoded.Bytes()...)
+
+		skipped := ptr.Int32(0)
+		ch := newWriterChannel(kafkaMessageTypeReplicationTask)
+
+		go func() {
+			err := parse(data, false, skipped, ch)
+			assert.NoError(t, err)
+			close(ch.ReplicationTaskChannel)
+		}()
+
+		result := <-ch.ReplicationTaskChannel
+
+		assert.NotNil(t, result)
+	})
+
+	t.Run("successful parse visibility task", func(t *testing.T) {
+		wireVal, _ := (&indexer.Message{Version: ptr.Int64(123)}).ToWire()
+		encoded := bytes.NewBuffer([]byte{})
+		_ = binary.Default.Encode(wireVal, encoded)
+		data := []byte("Partition: abc, Offset: 0, Key: 123")
+		data = append(data, preambleVersion0)
+		data = append(data, encoded.Bytes()...)
+
+		skipped := ptr.Int32(0)
+		ch := newWriterChannel(kafkaMessageTypeVisibilityMsg)
+
+		go func() {
+			err := parse(data, false, skipped, ch)
+			assert.NoError(t, err)
+			close(ch.VisibilityMsgChannel)
+		}()
+
+		result := <-ch.VisibilityMsgChannel
+
+		assert.NotNil(t, result)
+	})
+
+	t.Run("invalid replication task", func(t *testing.T) {
+		data := []byte("Partition: abc, Offset: 0, Key: 123")
+		data = append(data, preambleVersion0)
+		data = append(data, []byte{1, 2, 3, 4}...)
+
+		skipped := ptr.Int32(0)
+		ch := newWriterChannel(kafkaMessageTypeReplicationTask)
+
+		go func() {
+			err := parse(data, false, skipped, ch)
+			assert.EqualError(t, err, "parsing failed: Input was malformedError: unknown ttype Type(1)")
+			close(ch.ReplicationTaskChannel)
+		}()
+
+		result := <-ch.ReplicationTaskChannel
+
+		assert.Nil(t, result)
+	})
+
+	t.Run("invalid visibility task", func(t *testing.T) {
+		data := []byte("Partition: abc, Offset: 0, Key: 123")
+		data = append(data, preambleVersion0)
+		data = append(data, []byte{1, 2, 3, 4}...)
+
+		skipped := ptr.Int32(0)
+		ch := newWriterChannel(kafkaMessageTypeVisibilityMsg)
+
+		go func() {
+			err := parse(data, false, skipped, ch)
+			assert.EqualError(t, err, "parsing failed: Input was malformedError: unknown ttype Type(1)")
+			close(ch.VisibilityMsgChannel)
+		}()
+
+		result := <-ch.VisibilityMsgChannel
+
+		assert.Nil(t, result)
+	})
+}
+
+func TestAdminKafkaParse(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		defer func() {
+			_ = os.Remove("input.txt")
+			_ = os.Remove("output.txt")
+		}()
+
+		wireVal, _ := (&replicator.ReplicationTask{CreationTime: ptr.Int64(123)}).ToWire()
+		encoded := bytes.NewBuffer([]byte{})
+		_ = binary.Default.Encode(wireVal, encoded)
+		data := []byte("Partition: abc, Offset: 0, Key: 123")
+		data = append(data, preambleVersion0)
+		data = append(data, encoded.Bytes()...)
+
+		err := os.WriteFile("input.txt", data, 0666)
```

Comment:
I think it would be better to use createTempFileWithContent from the repo here (same for the output), same for the cases below